src/main/java/com/google/api/gax/core/FlowControlSettings.java

```diff
@@ -48,20 +47,33 @@ public static FlowControlSettings getDefaultInstance() {
   /** Maximum number of outstanding bytes to keep in memory before enforcing flow control. */
   @Nullable
   public abstract Integer getMaxOutstandingRequestBytes();
+  
+  /** Is flow control enabled. Defaults to true. */
+  public abstract boolean getIsEnabled();
+  
+  /**
+   * Control whether the FlowController will fail with an exception or block the thread when flow
+   * control limits are exceeded.
+   */
+  public abstract boolean getShouldFailOnFlowControlLimits();
```

Comment:
The name here doesn't make clear what happens if it is set to false. I think I would prefer an enum with the values `ThrowException` and `Block`, with this function renamed accordingly. I'm not sure what to call the enum - `FlowControlLimitBehavior` is a little long.